lib/core/plugins.rb

```diff
@@ -39,30 +41,48 @@ def load
   # If the plugin does not exist then it
   # @param Hash data
   # @return boolean - whether the command was passed on
-  def exec(data)
-    # if contains a space
-    if data.text.include? ' '
-      # Create array based on spaces
-      data_array = data.text.split(' ')
-      requested_plugin = data_array[1]
-      @plugin_hash.each do |name, plugin|
-        output = plugin.exec data if name == requested_plugin
-        return output if name == requested_plugin
-      end
+  def exec(requested_plugin, data)
+    @plugin_hash.each do |name, plugin|
+      output = plugin.exec data if name == requested_plugin
+      return output if name == requested_plugin
     end
-    nil
   end
 
   # Searches for phrased based plugins
+  # TODO: Create Phrases for Plugins: Create code to sift through chat data to match specific phrases for plugins
   def phrase_lookup
     # search plugin hash and container labels?
   end
 
   # Creates primary help list
   #
   # Utilizes the bot.yml help hash to determine response level.
-  def help
-    # Merge all hashes together or parse out a file(s)?
+  # TODO: Build Help Hash/Response to return after .each
+  def help(data)
+    if data.text.include? ' '
+      data_array = data.text.split(' ')
+      requested_plugin = data_array[2] unless data.channel[0] == 'D'
```

Comment:
A comment to explain the meaning of 'D' or point to documentation that explains it is likely needed here.